### ISSUE TITLE
[Bug] Workers produce identical augmentations per batch position across epochs when used with PyTorch DataLoader

### ISSUE THREAD

**Issue by Mateo-drr** [albumentations-team/AlbumentationsX#266]
### Environment:

albucore                0.0.24
albumentations          2.0.8
opencv-python-headless  4.13.0.92
torch: 2.11.0
Python: 3.14.4
OS: CachyOS

### Description

When using A.Compose inside a PyTorch Dataset.__getitem__ with num_workers > 0, each sample at a given batch position receives identical augmentations across epochs. The image content varies correctly (shuffle is working) but the augmentation applied — including randomized parameters like dropout position, shear amount, flip decisions — is identical for every image processed at the same intra-batch position.

The issue is resolved by setting num_workers=0.

Minimal reproducible example

```
import albumentations as A
import numpy as np
import torch
from torch.utils.data import Dataset, DataLoader
import matplotlib.pyplot as plt
import cv2
cv2.setNumThreads(0)

class TestDataset(Dataset):
    def __init__(self):
        self.augmentations = A.Compose([
            A.HorizontalFlip(p=0.5),
            A.Affine(p=0.5),
            A.CoarseDropout(num_holes_range=(1,1), p=0.5)
        ])
        self.data = np.ones((100, 32, 32, 3), dtype=np.uint8)*255

    def __len__(self):
        return len(self.data)

    def __getitem__(self, idx):
        img = self.augmentations(image=self.data[idx])['image']
        return torch.from_numpy(img).permute(2,0,1).float() / 255.0

if __name__ == "__main__":
    ds = TestDataset()
    dl = DataLoader(ds, batch_size=16, shuffle=True, num_workers=4)
    
    for runs in range(2):
        # Run two epochs and compare augmentations at position 0
        batches = []
        for epoch in range(2):
            for batch in dl:
                batches.append(batch)
                break
        
        print("Identical augmentation at position 0 across epochs:", 
              (batches[0][0] == batches[1][0]).all().item())
        
        plt.figure()
        plt.imshow(batches[0][0].permute(1,2,0))
        plt.show()
        plt.figure()
        plt.imshow(batches[1][0].permute(1,2,0))
        plt.show()
```

**Comment by lorenzo-morelli** [comment]
I noticed the same issue on my end.
python 3.14
albumentation 2.0.8
OS: Fedora Linux 43

**Comment by ternaus** [comment]
@Mateo-drr @lorenzo-morelli thank you for the report. Fixed. Should work correctly in version 2.3.1
